tests/features/unit-of-work/manual-pk-inverse-relation.test.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { Collection, Entity, ManyToOne, OneToMany, PrimaryKey, Rel, SimpleLogger } from '@mikro-orm/core';
+import { MikroORM, ObjectId, defineConfig } from '@mikro-orm/mongodb';
+import { v4 } from 'uuid';
+
+@Entity()
+class Foo {
+
+  @PrimaryKey()
+  public _id!: ObjectId;
+
+  @ManyToOne({ entity: () => Bar1, nullable: true })
+  public bar1?: Rel<Bar1>;
+
+  @ManyToOne({ entity: () => Bar2, nullable: true })
+  public bar2?: Rel<Bar2>;
+
+  @ManyToOne({ entity: () => Bar3, nullable: true })
+  public bar3?: Rel<Bar3>;
+
+  @ManyToOne({ entity: () => Bar4, nullable: true })
+  public bar4?: Rel<Bar4>;
+
+}
+
+@Entity()
+class Bar1 {
+
+  @PrimaryKey()
+  public _id!: ObjectId;
+
+  @OneToMany({ entity: () => Foo, mappedBy: book => book.bar1 })
+  public foos = new Collection<Foo>(this);
+
+}
+
+@Entity()
+class Bar2 {
+
+  @PrimaryKey()
+  public _id = v4();
+
+  @OneToMany({ entity: () => Foo, mappedBy: book => book.bar2 })
+  public foos = new Collection<Foo>(this);
+
+}
+
+@Entity()
+class Bar3 {
+
+  @PrimaryKey()
+  public _id!: string;
+
+  @OneToMany({ entity: () => Foo, mappedBy: book => book.bar3 })
+  public foos = new Collection<Foo>(this);
+
+}
+
+@Entity()
+class Bar4 {
+
+  @PrimaryKey()
+  public _id!: number;
+
+  @OneToMany({ entity: () => Foo, mappedBy: book => book.bar4 })
+  public foos = new Collection<Foo>(this);
+
+}
+
+let orm: MikroORM;
+let foo: Foo;
+
+beforeAll(async () => {
+  orm = await MikroORM.init(defineConfig({
+    entities: [
+      Foo,
+      Bar1,
+      Bar2,
+      Bar3,
+      Bar4,
+    ],
+    clientUrl: 'mongodb://db',
+    loggerFactory: options => new SimpleLogger(options),
+    debug: true,
+  }));
+});
+
+beforeEach(async () => {
+  await orm.schema.refreshDatabase();
+  foo = orm.em.create(Foo, {});
+  await orm.em.flush();
+  orm.em.clear();
```

Comment:
`foo` is detached entity, you need to reload it again after clearing, otherwise the whole test does not make much sense - such detached entity should be considered as new.

```suggestion
  foo = orm.em.create(Foo, {});
  await orm.em.flush();
//  orm.em.clear(); either this should be commented, or you need `em.findOne` call after it
```